### DISCUSSION TITLE
syntax error at pl/sql code

### DISCUSSION THREAD

**Discussion by abdullahameromran** [supabase/supabase#20392] [Questions] [answered]
Hi everybody .
what is the error here in my code ? 
![image](https://github.com/supabase/supabase/assets/113351133/267e0dbc-90a7-49f5-9068-aecfce58b4b9)

**Comment by GaryAustin1** [comment, marked as answer]
Did you use the AI to debug?
Your syntax looks odd.  You should have the variable name and then its type.  Then := val.   I don't know if you can do selects in the declare part or not.  You may have to do that after begin to set the value you want.
https://www.postgresql.org/docs/current/plpgsql-declarations.html

**Reply by abdullahameromran** [reply]
> Did you use the AI to debug? Your syntax looks odd. You should have the variable name and then its type. Then := val. I don't know if you can do selects in the declare part or not. You may have to do that after begin to set the value you want. https://www.postgresql.org/docs/current/plpgsql-declarations.html

Yes it works 
thanks @GaryAustin1 ;
